packages/web/pages/guild/[guildname]/edit.tsx

```diff
@@ -10,24 +10,112 @@ import {
 import { getGuild } from 'graphql/queries/guild';
 import { useRouter } from 'next/router';
 import Page404 from 'pages/404';
-import React, { useEffect, useState } from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import useSWR from 'swr';
+import { optimizedImage } from 'utils/imageHelpers';
 
 const EditGuild: React.FC = () => {
   const router = useRouter();
 
   const [guild, setGuild] = useState<GuildFragment | null>();
-  const [updateGuildState, updateGuild] = useUpdateGuildMutation();
+  const [{ fetching, data }, updateGuild] = useUpdateGuildMutation();
   const toast = useToast();
 
   const guildName = router.query.guildname as string;
 
-  const { data: fetchedGuild, isValidating } = useSWR(guildName, getGuild);
+  const { data: fetchedGuild, isValidating } = useSWR(guildName, getGuild, {
+    revalidateOnFocus: false,
+  });
 
   useEffect(() => {
     setGuild(fetchedGuild);
   }, [fetchedGuild, guildName]);
 
+  const onSubmit = useCallback(
+    async (editGuildFormInputs: EditGuildFormInputs) => {
+      if (!guild) return;
+
+      const {
+        type,
+        discordAdminRoles: adminRoles,
+        discordMembershipRoles: membershipRoles,
+        logoFile,
+        logoUrl,
+        ...otherInputs
+      } = editGuildFormInputs;
+
+      let newLogoUrl = logoUrl;
+
+      if (logoFile?.[0]) {
+        const formData = new FormData();
+        formData.append('logo', logoFile[0]);
+
+        const result = await fetch(`/api/storage`, {
+          method: 'POST',
+          body: formData,
+          credentials: 'include',
+        });
+
+        const response = await result.json();
+        const { error, logo } = response;
+
+        if (result.status >= 400 || error || !logo) {
+          const description =
+            result.status >= 400 || error
+              ? `web3.storage ${result.status} response: "${
+                  error ?? result.statusText
+                }"`
+              : `Uploaded logo but didn't get a response back.`;
+          toast({
+            title: 'Error Saving Logo',
+            description,
+            status: 'warning',
+            isClosable: true,
+            duration: 8000,
+          });
+          return;
+        }
+        newLogoUrl = optimizedImage('logoURL', `ipfs://${logo}`);
```

Comment:
Shouldn't we store the `ipfs` url rather than the processed version? Seems more future proof.

Reply:
fixed in [6426825](https://github.com/MetaFam/TheGame/pull/1367/commits/64268253048e84ba01f12253f02fb02efa03a6f4)